app/Http/Controllers/Api/DeviceController.php

```diff
@@ -318,5 +322,69 @@ public function anyList(BcsRequest $request)
 
         return response()->json($ret_val);
     }
-
+    
+    /**
+     * @param \BComeSafe\Http\Requests\SaveDeviceRequest $request
+     *
+     * @return array
+     */
+    public function anyUpdateDevice(SaveDeviceRequest $request)
+    {
+        $update = array();
+        $params = $request->all();
+        //
+        if (isset($params['user_phone'])) {
+          if (!empty($request->get('user_phone'))) {
+            $update['user_phone'] = $request->get('user_phone');
+            $update['need_phone'] = 0;
+            $update['user_phone_confirm'] = 0;
+            //Generate token
+            $update['user_phone_token'] = Device::generateToken();
+            //Send token via SMS
+            $defaults = SchoolDefault::getDefaults();
+            if (!empty($defaults->sms_provider)) {
```

Comment:
What happens for schools that don't have an SMS provider? Is `need_phone` set based on whether we are able to send SMS to the user?

Reply:
There is not relation with capability to send SMS.